guides/dimensiondata.md

```diff
@@ -0,0 +1,721 @@
+- Hardware - number of cores, RAM size and storage size for a given Image
+
+### Getting Started
+
+{% highlight java %}
+    ComputeService compute = ContextBuilder.newBuilder( "dimensiondata-cloudcontrol" )
+                             .credentials( "username", "password" )
+                             .buildView( ComputeServiceContext.class )
+                             .getComputeService();
+{% endhighlight %}                             
+
+The implementation of the jclouds abstraction is incomplete so it is best for now to use the unwrapped *DimensionDataCloudControlApi*. 
+
+> Take note that these features are still accessible by *unwrapping* the Dimension Data CloudControl API, but this'll reduce portability of your code. See [Concepts](https://jclouds.apache.org/start/concepts/).
+
+# How To's
+## <a id="how-to-list-a-data-center"></a>How to: List a Data Center
+
+The Dimension Data CloudControl organization will be associated with one or more [MCP 2.0 Data Centers](https://docs.mcp-services.net/display/CCD/Introduction+to+MCP+2.0+Data+Center+Locations). A Data Center is a container for the assets that will be deployed and created by the jclouds Dimension Data CloudControl Provider.
+
+In jclouds terminology a Zone equates to a Data Center.
+
+The following code example shows you how to programmatically list Data Centers: 
+
+{% highlight java %}
+     PagedIterable<Datacenter> datacenters = api.getInfrastructureApi().listDatacenters();
+{% endhighlight %}
+
+This responds with a paged result set containing the Data Centers visible to your organization.
+
+## <a id="how-to-deploy-a-network-domain"></a>How to: Deploy a Network Domain
+
+For more information on Network Domains see [here](https://docs.mcp-services.net/display/CCD/Introduction+to+Cloud+Network+Domains+and+VLANs)
+
+The code to deploy a network domain follows.
+
+{% highlight java %}
+        /*
+         * Deploy Network Domain to the Zone / Datacenter we wish to operate on. The response from this API is the Network Domain Identifier.
+         */
+        String networkDomainId = api.getNetworkApi().deployNetworkDomain(AU9, "jclouds-example", "jclouds-example", "ESSENTIALS");
+{% endhighlight %}
+
+A Network Domain deployment is an asynchronous process. We need to wait for it to complete. The Dimension Data provider
+has built in google guice predicates that will block execution and check that the Network Domain's State has moved from PENDING_ADD to NORMAL.
+
+Following is some example code that shows how the to use predicate suitable for asserting a Network Domain state has transitioned to the NORMAL state. The predicate uses the Network Domain Identifier we wish to check the state of.
+{% highlight java %}
+        Injector injector = contextBuilder.buildInjector();
```

Comment:
Don't do this. Instead use `context.utils().injector()` and if necessary explain how to build an `ApiContext< DimensionDataCloudControlApi>`. Users should never manually build the injector.

Reply:
Done